torch/linalg/__init__.py

```diff
@@ -483,6 +483,79 @@
     >>> tensor([ 11.6734+0.j, 105.1037+0.j,  10.1978+0.j])
 """)
 
+pinv = _add_docstr(_linalg.linalg_pinv, r"""
+linalg.pinv(input, rcond=1e-15, hermitian=False) -> Tensor
+
+Computes the pseudo-inverse (also known as the Moore-Penrose inverse) of a matrix :attr:`input`,
+or of each matrix in a batched :attr:`input`.
+The pseudo-inverse is computed using singular value decomposition (see :func:`torch.linalg.svd`) by default.
+If :attr:`hermitian` is ``True``, then :attr:`input` is assumed to be Hermitian (symmetric if real-valued),
+and the computation of the pseudo-inverse is done by obtaining the eigenvalues and eigenvectors
+(see :func:`torch.linalg.eigh`).
+The singular values (or the absolute eigenvalues when :attr:`hermitian` is ``True``) that are below
+the specified :attr:`rcond` threshold are treated to be zero and discarded in the computation.
+
+Supports input of ``float``, ``double``, ``cfloat`` and ``cdouble`` datatypes.
+
+.. note:: When given inputs on a CUDA device, this function synchronizes that device with the CPU.
+
+Args:
+    input (Tensor): the input matrix of size :math:`(m, n)` or the batch of matrices of size :math:`(*, m, n)`
+                    where `*` is one or more batch dimensions.
+    rcond (float, Tensor, optional): the tolerance value to determine the cutoff for small singular values. Default: 1e-15
```

Comment:
One challenge with clearly documenting rcond is explaining what different tensor shapes do. Do you think we can elaborate on this? In particular, the shape of rcond must be broadcastable to the singular value tensor returned from torch.svd, right? 

Is it worth including an rcond example where rcond isn't a scalar? Possibly by inspecting the singular value decomposition and then determining a cutoff?

Reply:
Yes, rcond must be broadcastable to the singular values. I think it's a good enough condition to mention and doesn't need further explanation?
Changed in https://github.com/pytorch/pytorch/pull/48399/commits/b8714d9241a2dbf6fad91e1316605066ad1e2085